build/Jamfile.v2

```diff
@@ -57,9 +54,18 @@ local log_setup_cxx_public_requirements = [ requires
         cxx11_lambdas
     ] ;
 
-project boost/log
+constant boost_dependencies_private :
+    /boost/align//boost_align
+    /boost/asio//boost_asio
+    /boost/interprocess//boost_interprocess
+    /boost/predef//boost_predef
+    /boost/random//boost_random ;
+
+project
     : source-location ../src
+    : common-requirements <library>$(boost_dependencies)
```

Comment:
I think, dependency `requirements` and `usage-requirements` need to be specified per-library rather than for the project, as these requirements are different.